backend/src/main/java/com/github/NicolaiKopka/users/SpotifyLoginService.java

```diff
@@ -0,0 +1,23 @@
+package com.github.NicolaiKopka.users;
+
+import com.github.NicolaiKopka.db_models.spotifyModels.authenticationModels.SpotifyUserData;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SpotifyLoginService {
+
+    private final MyUserRepo myUserRepo;
+
+    public MyUser findOrCreateUser(SpotifyUserData userData) {
+        return myUserRepo.findByUsername(userData.getUsername()).orElseGet(() -> myUserRepo.save(MyUser.builder()
+                .username(userData.getUsername())
+                .email(userData.getEmail())
+                .roles(List.of("user"))
+                .build()));
+    }
```

Comment:
Is there any fallback if a registered User( from Username/Password registration) has a matching username with a Spotify-Register-User?

Reply:
good point. Still thinking about that problem.